### ISSUE TITLE
XS_SDL__Video_set_color_key () related bug

### ISSUE THREAD

**Issue by varnie** [PerlGameDev/SDL#248]
Hey.
I've got "Segmentation fault (core dumped)" when I intentionally improperly used "SDL::Video::set_color_key", i.e.:

```
SDL::Video::set_color_key(
    $m_surface, 
    SDL_SRCCOLORKEY | SDL_RLEACCEL, 
    SDL::Video::map_RGB(
        1, #this line is buggy!!! it should be $m_surface_new->format to be OK
        0xF1,
        0xCB,
        0x86
));
```

Running core-file under gdb gives the following output:
"
varnie@localhost:~/project $ gdb perl core 
GNU gdb (GDB) 7.5-ubuntu
Copyright (C) 2012 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later http://gnu.org/licenses/gpl.html
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "i686-linux-gnu".
For bug reporting instructions, please see:
http://www.gnu.org/software/gdb/bugs/...
Reading symbols from /usr/bin/perl...(no debugging symbols found)...done.

warning: core file may not match specified executable file.
[New LWP 11325]

warning: Can't read pathname for load map: Input/output error.
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/i386-linux-gnu/libthread_db.so.1".
Core was generated by `/usr/bin/perl ./yup.pl'.
Program terminated with signal 11, Segmentation fault.
#0  0xb69f99a4 in XS_SDL__Video_set_color_key ()

   from /usr/lib/perl5/auto/SDL/Video/Video.so
(gdb) 
"
In other words there's definitely a bug in the 'XS_SDL__Video_set_color_key ()' function (wrong handling of incorrect parameters or alike).

I am running Perl v5.14.2 with libsdl-perl v2.540-1 and libalien-sdl-perl v1.430-4 if it helps.
